src/panels/config/users/ha-config-users.ts

```diff
@@ -0,0 +1,192 @@
+import "../../../layouts/hass-tabs-subpage-data-table";
+import "../../../components/ha-fab";
+
+import { computeRTL } from "../../../common/util/compute_rtl";
+import { configSections } from "../ha-panel-config";
+import {
+  LitElement,
+  property,
+  css,
+  PropertyValues,
+  customElement,
+} from "lit-element";
+import { HomeAssistant, Route } from "../../../types";
+import { html } from "lit-html";
+import { HASSDomEvent } from "../../../common/dom/fire_event";
+import { User, fetchUsers, updateUser, deleteUser } from "../../../data/user";
+import memoizeOne from "memoize-one";
+import {
+  DataTableColumnContainer,
+  RowClickedEvent,
+} from "../../../components/data-table/ha-data-table";
+import { showUserDetailDialog } from "./show-dialog-user-detail";
+import { showAddUserDialog } from "./show-dialog-add-user";
+import { showConfirmationDialog } from "../../../dialogs/generic/show-dialog-box";
+
+@customElement("ha-config-users")
+export class HaConfigUsers extends LitElement {
+  @property() public hass!: HomeAssistant;
+  @property() public _users: User[] = [];
+  @property() public isWide!: boolean;
+  @property() public narrow!: boolean;
+  @property() public route!: Route;
+
+  private _columns = memoizeOne(
+    (_language): DataTableColumnContainer => {
+      return {
+        name: {
+          title: this.hass.localize(
+            "ui.panel.config.users.picker.headers.name"
+          ),
+          sortable: true,
+          filterable: true,
+          direction: "asc",
+          grows: true,
+          template: (name) => html`
+            ${name ||
+              this.hass!.localize("ui.panel.config.users.editor.unnamed_user")}
+          `,
+        },
+        group_ids: {
+          title: this.hass.localize(
+            "ui.panel.config.users.picker.headers.group"
+          ),
+          sortable: true,
+          filterable: true,
+          width: "25%",
+          template: (groupIds) => html`
+            ${this.hass.localize(`groups.${groupIds[0]}`)}
+          `,
+        },
+        system_generated: {
+          title: this.hass.localize(
+            "ui.panel.config.users.picker.headers.system"
+          ),
+          type: "icon",
+          width: "10%",
+          sortable: true,
+          filterable: true,
+          template: (generated) => html`
+            ${generated
+              ? html`
+                  <ha-icon icon="hass:check-circle-outline"></ha-icon>
+                `
+              : ""}
+          `,
+        },
+      };
+    }
+  );
+
+  protected firstUpdated(changedProperties: PropertyValues) {
+    super.firstUpdated(changedProperties);
+    this._fetchUsers();
+  }
+
+  protected render() {
+    return html`
+      <hass-tabs-subpage-data-table
```

Comment:
We should consider filtering out system users automatically. And also hide that column.